src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnector.java

```diff
@@ -0,0 +1,166 @@
+import io.cdap.cdap.etl.api.connector.BrowseEntity;
+import io.cdap.cdap.etl.api.connector.BrowseRequest;
+import io.cdap.cdap.etl.api.connector.Connector;
+import io.cdap.cdap.etl.api.connector.ConnectorContext;
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.ConnectorSpecRequest;
+import io.cdap.cdap.etl.api.connector.DirectConnector;
+import io.cdap.cdap.etl.api.connector.PluginSpec;
+import io.cdap.cdap.etl.api.connector.SampleRequest;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.plugin.common.ConfigUtil;
+import io.cdap.plugin.common.Constants;
+import io.cdap.plugin.common.ReferenceNames;
+import io.cdap.plugin.sfmc.sink.MarketingCloudDataExtensionSink;
+import io.cdap.plugin.sfmc.source.MarketingCloudClient;
+import io.cdap.plugin.sfmc.source.MarketingCloudInputSplit;
+import io.cdap.plugin.sfmc.source.MarketingCloudSource;
+import io.cdap.plugin.sfmc.source.MarketingCloudSourceConfig;
+import io.cdap.plugin.sfmc.source.util.MarketingCloudConstants;
+import io.cdap.plugin.sfmc.source.util.SourceObject;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Salesforce Marketing Cloud Connector
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(MarketingCloudConstants.PLUGIN_NAME)
+@Description("Connection to access data from Salesforce Marketing Objects.")
+
+public class MarketingConnector implements DirectConnector {
+
+    private MarketingConnectorConfig config;
+
+    private static final String ENTITY_TYPE_OBJECTS = "object";
+    private ETResponse<? extends ETSoapObject> response;
+    private Schema schema;
+    public ETApiObject row;
+
+    private StructuredRecord record;
+    private Iterator<? extends ETApiObject> iterator;
+
+    private List<? extends ETApiObject> results;
+
+    private String requestId;
+
+    private String dataExtensionKey = "";
+
+
+    @Override
+    public void test(ConnectorContext connectorContext) throws ValidationException {
+        FailureCollector collector = connectorContext.getFailureCollector();
+        config.validateCredentials(collector);
+    }
+
+    @Override
+    public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+        BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+        int count = 0;
+        try {
+            for (SourceObject object : SourceObject.values()) {
+                String name = object.getValue();
+                BrowseEntity.Builder entity = (BrowseEntity.builder(String.valueOf(object), String.valueOf(object),
+                                ENTITY_TYPE_OBJECTS).
+                        canBrowse(false).canSample(true));
+                browseDetailBuilder.addEntity(entity.build());
+                count++;
+            }
+        } catch (Exception e) {
```

Comment:
You can remove this try catch as we are not making any API calls

Reply:
Yes. Makes Sense. I have already removed it.